.github/workflows/lint.yml

```diff
@@ -9,16 +9,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: astral-sh/setup-uv@v5
+      - run: uv run ruff check --no-fix --select PLE
       - run: uv run ruff format
       - run: uv run pre-commit run --all-files
-      # TODO: Fix the ignored pytests.
-      # openai.OpenAIError: The api_key client option must be set either by passing
-      # api_key to the client or by setting the OPENAI_API_KEY environment variable
-      - run: uv run --with=dotenv pytest
-                --ignore=tests/test_dropdown_error.py
-                --ignore=tests/test_gif_path.py
-                --ignore=tests/test_models.py
-                --ignore=tests/test_react_dropdown.py
-                --ignore=tests/test_save_conversation.py
-                --ignore=tests/test_vision.py
-                --ignore=tests/test_wait_for_element.py || true
+      - run: uv run pyright
```

Comment:
lets split this into 3 separate jobs (up a level from here) so they get separate rows in the Github PR UI and so they can be parallelized (it saves a lot of clicks having to dig into CI errors every time to see what failed).

- cheap and fast check for basic syntax errors w/ PLE
- check for ruff style/formatting + pre-commit hook errors
- check for errors from type checker (pyright)

all 3 can share the same uv cache so they should each run for <20sec in theory, and if they run in parallel then all PR checks can be ready in <30sec.